lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -339,18 +452,38 @@ impl ChunkedGelfDecoder {
         &mut self,
         mut src: Bytes,
     ) -> Result<Option<Bytes>, ChunkedGelfDecoderError> {
-        if src.starts_with(GELF_MAGIC) {
+        let message = if src.starts_with(GELF_MAGIC) {
+            debug!("Received a chunked GELF message based on the magic bytes");
             src.advance(2);
-            self.decode_chunk(src)
+            self.decode_chunk(src)?
         } else {
-            Ok(Some(src))
-        }
+            debug!(
+                "Received an unchunked GELF message. First two bytes of message: {:?}",
+                &src[0..2]
+            );
+            Some(src)
+        };
+
+        // We can have both chunked and unchunked messages that are compressed
+        message
+            .map(|message| {
```

Comment:
Right know, the decompression is a bit coupled with the chunked gelf decoder. 

I thought about adding the automatic decompression at a generic point such as https://github.com/vectordotdev/vector/blob/983a9937ac5fc24db0203eec0390550351ec2cbe/src/codecs/decoding/decoder.rs#L91. This would result in adding an additional step between framing and format deserializing that would involve this decompression part. This would decouple the decompression from the underlying framing, and I think it would be useful for reading compressed bytes from _any_ source, without requiring to implement the decoding for every source (for example, kafka, file source, etc).

What do you think? Is it worth the decoupling? I also like to leave this here as it is, coupled with the chunked gelf framer.

Reply:
> I think it would be useful for reading compressed bytes from any source, without requiring to implement the decoding for every source (for example, kafka, file source, etc).

Agreed. This can be a followup PR though since it might be tricky to test.